starlark/testdata/math.star

```diff
@@ -17,16 +17,24 @@ assert.eq(math.ceil(10.0), 10.0)
 assert.eq(math.ceil(0), 0.0)
 assert.eq(math.ceil(1), 1.0)
 assert.eq(math.ceil(10), 10.0)
-assert.eq(math.ceil(inf), inf)
-assert.eq(math.ceil(nan), nan)
 assert.eq(math.ceil(-0.0), 0.0)
 assert.eq(math.ceil(-0.4), 0.0)
 assert.eq(math.ceil(-0.5), 0.0)
 assert.eq(math.ceil(-1.0), -1.0)
 assert.eq(math.ceil(-10.0), -10.0)
 assert.eq(math.ceil(-1), -1.0)
 assert.eq(math.ceil(-10), -10.0)
-assert.eq(math.ceil(-inf), -inf)
+assert.eq(type(math.ceil(0)), "int")
+assert.eq(type(math.ceil(0.4)), "int")
+assert.eq(type(math.ceil(10)), "int")
+assert.eq(type(math.ceil(-10.0)), "int")
+assert.eq(type(math.ceil(-0.5)), "int")
+assert.fails(
+  lambda: math.ceil(inf), "cannot convert float infinity to integer")
+assert.fails(
+  lambda: math.ceil(-inf), "cannot convert float infinity to integer")
+assert.fails(
+  lambda: math.ceil(nan), "cannot convert float NaN to integer")
```

Comment:
Let's add a test for a float value larger than int64.

Reply:
So, if the float value is significantly larger than `math.MaxInt64`, then the function works as expected, e.g.

```python
f = 1.8446744073709552e+19
assert.eq(math.ceil(f), f)  # ok
```

However, if it is slightly larger, then we'll get an overflow:

```python
q = 9223372036854775807.5
print(math.ceil(q))  # -9223372036854775808
```

This seems to be because `q` is close enough to `math.MaxInt64` that the [check in `finiteFloatToInt` will register it as an int](https://github.com/google/starlark-go/blob/master/starlark/int.go#L427).

Is this ok? Or should the `ceil` function take this into consideration?

Reply:
See https://github.com/google/starlark-go/pull/376/files